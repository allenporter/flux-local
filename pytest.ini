[pytest]
log_level = DEBUG
asyncio_mode = auto
