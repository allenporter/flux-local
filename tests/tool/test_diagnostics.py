"""Tests for the flux-local `diagnostics` command."""

import pathlib
import tempfile

from syrupy.assertion import SnapshotAssertion
import yaml
import pytest

from . import run_command


async def test_diagnostics(snapshot: SnapshotAssertion) -> None:
    """Test test get ks commands."""
    result = await run_command(["diagnostics"])
    assert result == snapshot


@pytest.mark.parametrize(
    ("contents"),
    [
        (yaml.dump_all([["entry"]]).encode()),
        (yaml.dump_all(["entry", "var"]).encode()),
        (b"\x00\x00\x00"),
    ],
    ids=["list", "scalar", "not-yaml"],
)
async def test_invalid_mapping(contents: bytes, snapshot: SnapshotAssertion) -> None:
    """Test test get ks commands."""
    with tempfile.TemporaryDirectory() as tmp_dir:
        example_file = pathlib.Path(tmp_dir) / "example.yaml"
        example_file.write_bytes(contents)

        result = await run_command(["diagnostics", "--path", tmp_dir])
        result = result.replace(str(example_file), "<<TEST_FILE>>")
        assert result == snapshot
