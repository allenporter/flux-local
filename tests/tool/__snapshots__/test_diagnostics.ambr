# serializer version: 1
# name: test_diagnostics
  '''
  [DIAGNOSTICS OK]
  
  '''
# ---
# name: test_invalid_mapping[list]
  '''
  [DIAGNOSTICS FAIL]: `<<TEST_FILE>>` expected dictionary but was list: ['entry']
  
  '''
# ---
# name: test_invalid_mapping[not-yaml]
  '''
  [DIAGNOSTICS FAIL]: `<<TEST_FILE>>` failed to parse as yaml: unacceptable character #x0000: special characters are not allowed
    in "<unicode string>", position 0
  
  '''
# ---
# name: test_invalid_mapping[scalar]
  '''
  [DIAGNOSTICS FAIL]: `<<TEST_FILE>>` was not a dictionary: <class 'str'>: entry
  [DIAGNOSTICS FAIL]: `<<TEST_FILE>>` was not a dictionary: <class 'str'>: var
  
  '''
# ---
